### PR TITLE
Add CSV exports to the dashboard

### PR DESCRIPTION
**Pull request by davidbgk**

### CODE REVIEW

File: croquemort/reports.py [hunk 1]
```diff
@@ -94,3 +97,45 @@ def compute_report(urls, filters, excludes, querystring, with_links=False):
...
+        fake_file.truncate(0)
+
+        for url_hash, data in urls:
+            # Filtering.
+            data = apply_filters(data, filters, excludes)
+            if not data:
+                continue
+
+            # Statuses.
+            status = data.get('status', '')
```
Comment (noirbizarre) [inline croquemort/reports.py]: Why not simply `data.get('status')` ?

File: croquemort/reports.py [hunk 2]
```diff
@@ -94,3 +97,45 @@ def compute_report(urls, filters, excludes, querystring, with_links=False):
...
+            if not status:
+                continue
+
+            url = data['url']
+            content_type = data.get('content-type', '')
+            updated = data.get('updated', '')
```
Comment (noirbizarre) [inline croquemort/reports.py]: same as above for these two

File: croquemort/reports.py [hunk 3]
```diff
@@ -94,3 +97,45 @@ def compute_report(urls, filters, excludes, querystring, with_links=False):
...
+
+            w.writerow((url, status, content_type, updated))
+            yield fake_file.getvalue()
+            fake_file.seek(0)
+            fake_file.truncate(0)
+
+    # Set headers with the appropriated filename.
+    headers = Headers()
+    headers.set('Content-Disposition', 'attachment', filename='croquemort.csv')
```
Comment (noirbizarre) [inline croquemort/reports.py]: It might be relevant to timestamp the filename

**Comment by noirbizarre** [comment]
r+wc

File: croquemort/reports.py (hunk 1 above)
Comment (davidbgk) [inline croquemort/reports.py]: Otherwise it returns `None` which is not a valid CSV entry.

File: croquemort/reports.py (hunk 2 above)
Comment (davidbgk) [inline croquemort/reports.py]: Same answer :)

File: croquemort/reports.py (hunk 3 above)
Comment (davidbgk) [inline croquemort/reports.py]: Good idea!

File: croquemort/reports.py (hunk 1 above)
Comment (karlcow) [inline croquemort/reports.py]: Just passing by

```python
>>> data = {'url_1': '123', 'url_2': '2222'}
>>> data.get('url_1')
'123'
>>> data['url_1']
'123'
>>> data.get('url_3')
>>> data.get('url_3', '')
''
>>> data['url_3']
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
KeyError: 'url_3'
```

the `get` says, if the key is not here please use `None` or this value instead.

Comment (noirbizarre) [inline croquemort/reports.py]: Thanks (but I know how `dict.get()` works), I just saw the test after but I didn't saw the value is serialized into the CSV output and so using the empty string as default value makes sense now ;)

Comment (karlcow) [inline croquemort/reports.py]: yup not for you. For anyone coming after and reading.

Comment (noirbizarre) [inline croquemort/reports.py]: I know ;) I was explaining my misreading ;)

**Pull request merged**